src/client/datascience/interactive-window/interactiveWindow.ts

```diff
@@ -275,10 +276,14 @@ export class InteractiveWindow implements IInteractiveWindowLoadable {
                 ).controller.onDidChangeSelectedNotebooks(
                     (selectedEvent: { notebook: NotebookDocument; selected: boolean }) => {
                         // Controller was deselected for this InteractiveWindow's NotebookDocument
-                        if (selectedEvent.selected === false && selectedEvent.notebook === notebookDocument) {
+                        if (
+                            selectedEvent.selected === false &&
+                            selectedEvent.notebook.uri.toString() === currentNotebookUri
```

Comment:
This change isn't necessary (i.e. we should be able to compare notebook objects)

Reply:
When debugging this locally I found that we were failing the comparison check even when there was only one NotebookDocument (the embedded one in the IW). I don't want to rely on the NotebookDocument not changing over time for this code to work correctly

Reply:
Given that was the reason for the string compare, can you put a comment to that effect? So that the next person looking at this doesn't go - oh we don't need to compare strings.

Reply:
Or maybe a new function like ```doesNotebookMatch``` that has that comment in it and use that in the two places where you do the string compare?

Reply:
> When debugging this locally I found that we were failing the comparison check even when there was only one NotebookDocument (the embedded one in the IW). I don't want to rely on the NotebookDocument not changin

This is a bug in VS Code. it should be the same (everywhere in our extension we use notebook documents as keys in dictionaries).
@rebornix Any idea what's going on?

Reply:
Because if we cannot rely on notebook document being the same, then we should fix this in vscode or we should change the rest of our code

Reply:
It seems like a bug in VS Code, we should be able to compare notebook document as they are fat objects and always one document object per resource in one extension host.

Reply:
Just check with @rebornix it should be the same, & we should be using object ref comparisons.

Reply:
Updated my comment from `just check` to `just checked`